flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/AbstractMetricGroup.java

```diff
@@ -169,19 +176,7 @@ public String getMetricIdentifier(String metricName) {
 	 * @return fully qualified metric name
 	 */
 	public String getMetricIdentifier(String metricName, CharacterFilter filter) {
-		if (scopeString == null) {
-			if (filter != null) {
-				scopeString = ScopeFormat.concat(filter, registry.getDelimiter(), scopeComponents);
-			} else {
-				scopeString = ScopeFormat.concat(registry.getDelimiter(), scopeComponents);
-			}
-		}
-
-		if (filter != null) {
-			return scopeString + registry.getDelimiter() + filter.filterCharacters(metricName);
-		} else {
-			return scopeString + registry.getDelimiter() + metricName;
-		}
+		return getMetricIdentifier(metricName, filter, -1);
```

Comment:
if reporter index not [correct for array](https://github.com/apache/flink/pull/2650/files#diff-3cbfc4ef461f4bac5ddd197c77fe19b1R191) then in this block not use ``reporterIndex`` and call `MetricRegistry#getDelimiter()`
I will change to
``` java
if (scopeStrings.length == 0 || (reporterIndex < 0 || reporterIndex >= scopeStrings.length)) {
			if (filter != null) {
				String newScopeString = ScopeFormat.concat(filter, registry.getDelimiter(), scopeComponents);
				return newScopeString + registry.getDelimiter() + filter.filterCharacters(metricName);
			} else {
				String newScopeString = ScopeFormat.concat(registry.getDelimiter(), scopeComponents);
				return newScopeString + registry.getDelimiter() + metricName;
			}
		}
```
and warning message will not show